app/src/lib/stores/app-store.ts

```diff
@@ -2379,7 +2379,8 @@ export class AppStore extends TypedBaseStore<IAppState> {
     return this.withIsCommitting(repository, async () => {
       const result = await gitStore.performFailableOperation(async () => {
         const message = await formatCommitMessage(repository, context)
-        return createCommit(repository, message, selectedFiles) !== undefined
+        await createCommit(repository, message, selectedFiles)
+        return true
```

Comment:
Since `createCommit` can't be `undefined` anymore, I simplified this to just return `true` when it doesn't fail (otherwise, `result` will be `undefined` thanks to `performFailableOperation`). Any suggestions to make this cleaner?

Reply:
> and then check if result is not undefined (as opposed to the truthy check that's in place right now

That's what we had before (since #11345, more specifically). But with the changes in this PR, `createCommit` will never return `undefined`, since that could only happen when throwing an exception and `performFailableOperation` would handle that one.

I changed it because it felt wrong / dishonest, meaning it could lead you to think it returns `undefined` but it doesn't.

Anyway, happy to go ahead with the `!== undefined` if you find it cleaner!

Reply:
Hopefully this looks ok! f7efae756426670ffcea26981d0580d4ce2f8a51